powerauth-push-server/src/main/java/io/getlime/push/service/CaCertificateService.java

```diff
@@ -59,12 +58,21 @@ public class CaCertUtil {
 
     private final ResourceLoader resourceLoader;
 
+    private List<X509Certificate> allCertificates;
+
     /**
      * Obtain all registered CA certificates.
      * @return All registered CA certificates.
      */
     public X509Certificate[] allCerts() {
-        // Prepare result list
+        return allCertificates.toArray(X509Certificate[]::new);
+    }
+
+    /**
+     * Load the locally stored CA certificates required by Apple for APNs.
+     */
+    @PostConstruct
```

Comment:
@zcgandcomp This may affect startup time, but the benefit is, that it is loaded once per push-server instance and fails early if any error. This used to be called per APNS client and failed when called.